### ISSUE TITLE
Dropdown with description needed

### ISSUE THREAD

**Issue by yih-wang** [patternfly/patternfly-design#770]
We need a dropdown with description in it in AMQ. So users can have some help information about the options. I created an initial wireframe like this.
<img width="343" alt="Screen Shot 2019-05-17 at 15 47 52" src="https://user-images.githubusercontent.com/23331885/57911749-29fa5580-78bb-11e9-93c7-e88043008b11.png">

**Comment by mcarrano** [comment]
Should consider this as we work on #789

**Comment by mmenestr** [comment]
Here are two options I mocked up, one with the description using Red Hat text 14 sized font, and one with the description using Red Hat text 12. From the initial mockup it seems like size 12 font would be better but wanted to show both options! @mcarrano @gdoyle1 let me know your thoughts!

![image](https://user-images.githubusercontent.com/51165119/80154883-74398b80-858e-11ea-877b-d4cf2f0f81bc.png)

**Comment by mcarrano** [comment]
I agree with your assessment @mmenestr . Do we use 12pt type anywhere else? Something we should explore is what the hover state of the menu item looks like. I'm concerned that the gray text will not be readable against anything but white. Maybe that doesn't matter? What do you think?

**Comment by mmenestr** [comment]
I believe that the 12pt font is used in chips as well, not sure beyond that. It is a text style in the symbol library so we wouldn't need to introduce a new text style if that's what you're asking!

That's a great point about the hover state, I just drafted what it would look like using the current dropdown hover state and colors:

![image](https://user-images.githubusercontent.com/51165119/80236946-e1e6c580-8629-11ea-8bfe-de5f01cf6826.png)

It is readable, and it seems that it passes accessibility as well!
<img width="472" alt="Screen Shot 2020-04-24 at 12 43 28 PM" src="https://user-images.githubusercontent.com/51165119/80237091-1f4b5300-862a-11ea-8665-232745763e67.png">

**Comment by gdoyle1** [comment]
@mmenestr I think this looks great! Yes 12px is used for chips, so we're good there. Curious what the spacing is between the two text boxes and how it would look if the larger text went to two lines. And then lastly how it looks if the menu item is disabled!

**Comment by mmenestr** [comment]
Here are all the possible variations:
- All select, hover and disabled states are possible and pass accessibility.
- @gdoyle1 , we just established the component looks like it doesn't support text text going beyond two lines, but I included what that might look like anyway... Would be nice if at a minimum the description could span two lines? What do you think? @mcarrano 
- Lastly, I mocked up two versions of the select. One where the select is aligned with the title (as it usually would be) and one where the select is center aligned in the select box. I suspect it would be easier to just keep it aligned with the title, but just wanted to put both options out there!

![image](https://user-images.githubusercontent.com/51165119/80379804-996b1a00-886c-11ea-8a8e-fe8bc84453af.png)

**Comment by gdoyle1** [comment]
@mmenestr Thanks for going through all of these! I like the check top-aligned with the menu item. It looks like the select and dropdown components don't support multi-lined menu items _but_ I agree with you that it'd be nice if at least the description were to be able to support two lines!

**Comment by mcarrano** [comment]
These look good @mmenestr . I believe that the way this is currently constructed is that menu items do not wrap, but the width of the menu is set by the longest item.  What I would then expect is that we keep that behavior but let the description text wrap if it needs to, i.e. is longer that the menu item text. Can you confirm that this makes sense @mcoker ?

I agree with @gdoyle1 about top aligned placement of the checkmark.

@yih-wang @itsptk can you also take a look and make sure the proposal here meets your project needs? Sorry it's taken so long to address this, but it was something that just got lost in the backlog.

**Comment by itsptk** [comment]
@mcarrano The particular design I was mentioning was never implemented, but if it was I think this proposal could be useful. I also see this improvement as probably being useful for future use cases in general. Thanks for following up.

**Comment by mcoker** [comment]
> What I would then expect is that we keep that behavior but let the description text wrap if it needs to, i.e. is longer that the menu item text. Can you confirm that this makes sense @mcoker ?

@mcarrano that is my expectation as well.

**Comment by mmenestr** [comment]
Final specs!

![image](https://user-images.githubusercontent.com/51165119/80751167-52dd2000-8af7-11ea-85d1-d638688727b4.png)

**Comment by mcarrano** [comment]
Thanks for your work on this @mmenestr . I have opened core issue https://github.com/patternfly/patternfly/issues/3057 to implement these changes.
